src/Microsoft.VisualStudio.Web.CodeGeneration.EntityFrameworkCore/PropertyMetadata.cs

```diff
@@ -62,7 +62,10 @@ public PropertyMetadata(IProperty property, Type dbContextType)
 
             IsEnum = property.ClrType.GetTypeInfo().IsEnum;
             IsReadOnly = property.IsReadOnlyBeforeSave;
-            IsAutoGenerated = property.IsStoreGeneratedAlways;
+
+            IsAutoGenerated = ((property.RequiresValueGenerator
+                || property.IsStoreGeneratedAlways)
+                && (property.ValueGenerated != ValueGenerated.Never));
```

Comment:
How is `IsAutoGenerated` used? Values can be generated when entities start being tracked, and values can be generated/changed when an entity is saved to the database. However, in neither case is a value necessarily generated--usually if a value has already been set by the application, then it will be used instead of generating a new value.